src/assisted_test_infra/test_infra/controllers/minikube.py

```diff
@@ -0,0 +1,242 @@
+        )
+        # In case the scale increased check if all pods are ready and active
+        if replicas_count:
+
+            def pods_are_running():
+                # Waiter lambda verify all pods are running begins with the same name
+                pods = self._get_obj_resources_by_name(self.POD, name)
+                running = list(map(lambda pod: hasattr(pod.status, "phase") and pod.status.phase == "Running", pods))
+                return all(running)
+
+            waiting.wait(
+                lambda: pods_are_running() is True,
+                timeout_seconds=timeout,
+                sleep_seconds=sleep,
+                waiting_for="replicas to change successfully",
+            )
+        time.sleep(recover_time_pods)
+
+    def change_deployment_scale(self, deployment_name, replicas_count):
+        """Common code for scale up and down deployment
+        We assume that assisted-service name is unique in deployment. other kinds
+        name starts with same name with uuids like assisted-service-9fdd7f47-xd8vz
+        :param deployment_name: assisted-service
+        :param replicas_count: total number , 0 for stop 1 and more for start
+        :return:
+        """
+        deployment = self._get_obj_resources_by_name(self.DEPLOYMENT, deployment_name)
+        # verify single name for deployment replicaset exists
+        assert len(deployment) == 1, "deployment kind name should be unique"
+        replicaset = self._get_obj_resources_by_name(self.REPLICASET, deployment_name)
+        assert len(replicaset) == 1, "replicaset kind name should be unique"
+        deployment_name = deployment[0].metadata.name
+        body = {"spec": {"replicas": replicas_count}}
+        # Patch change configuration
+        self.patch_resource(self.DEPLOYMENT, deployment_name, body)
+        self.wait_for_scale_config_change(self.DEPLOYMENT, deployment_name, replicas_count)
+        self.wait_for_scale_config_change(self.REPLICASET, deployment_name, replicas_count)
+        # When changed to zero pods with this name should be deleted.
+        self.wait_for_pod_change(deployment_name, replicas_count)
+
+    def change_configmap_data(self, dict_data, restart_deployment=True, deployment_name="assisted-service"):
+        """Accept data we would like to change and patch it to configmap
+        Example:
+        {data: {'ENABLE_UPGRADE_AGENT': 'True', 'AGENT_DOCKER_IMAGE': agent_docker_image + tag_version}}
+        :param restart_deployment after configuration change
+        :param deployment_name - the deployment name to scale down/up
+        :param dict_data:
+        :return:
+        """
+        body = dict_data
+        self.patch_resource(self.CONFIGMAP, self.configmap_name, body)
+        # stop /start development for service
+        if restart_deployment:
+            self.change_deployment_scale(deployment_name, 0)
+            self.change_deployment_scale(deployment_name, 1)
```

Comment:
Maybe make a function out of it? You are doing the same logic also in `rollout_assisted_service`